src/runtime/arc.ts

```diff
@@ -93,11 +93,11 @@ export class Arc {
   loadedParticleInfo = new Map<string, {spec: ParticleSpec, stores: Map<string, StorageProviderBase>}>();
   readonly pec: ParticleExecutionHost;
 
-  constructor({id, context, pecFactory, slotComposer, loader, storageKey, storageProviderFactory, speculative, innerArc, stub, inspectorFactory, ports} : ArcOptions) {
+constructor({id, context, pecFactories, slotComposer, loader, storageKey, storageProviderFactory, speculative, innerArc, stub, inspectorFactory} : ArcOptions) {
     // TODO: context should not be optional.
     this._context = context || new Manifest({id});
-    // TODO: pecFactory should not be optional. update all callers and fix here.
-    this.pecFactory = pecFactory || FakePecFactory(loader).bind(null);
+    // TODO: pecFactories should not be optional. update all callers and fix here.
+    this.pecFactories = pecFactories && pecFactories.length > 0 ? pecFactories.slice() : [FakePecFactory(loader).bind(null)];
```

Comment:
`FakePecFactory` could also be an industry instead.

Reply:
i'll send a follow up pr